README.md

```diff
@@ -1,16 +1,226 @@
+
+```ruby
+2.3.3 :002 > "hi".class
+#=> String
+2.3.3 :002 > "there".class
+#=> String
+```
+
+We expect that if a `String` can do something, any `String` can do that similar
+thing, that's a core part of what makes similar things similar in our human
+brains.
+
+For example, in Ruby, `String`s know how long they are. We'll explain what's
+going on specifically in the next section, but....
+
+```ruby
+2.3.3 :003 > "hi".length
+#=> 2
+2.3.3 :004 > x = "hi"
+#=> "hi"
+2.3.3 :005 > x.length
+#=> 2
+2.3.3 :006 > y = "there"
+#=> "there"
+2.3.3 :007 > y.length
+#=> 5
+2.3.3 :008 > tip = "These two strings need at least #{ x.length + y.length } slots"
+#=> "These two strings need at least 7 slots"
+2.3.3 :009 > puts tip
+These two strings need at least 7 slots
+#=> nil
+```
+
+Amazing, right? Instances being smart about themselves means you can ask them
+what they know about themselves and don't have to do the work yourself! Ruby's
+default classes (`String`, `Integer` and many others!) can do a lot of work for
+you so that you can focus on your dreams instead of writing "external" methods
+to count `String` length.
+
+## Define Attribute / Property
+
+An attribute on an instance is a fixed bit of information that the instance can
+provide you. In Ruby, all instances have an attribute called `.class`. `String`s
+in Ruby also have the attribute or property called `length`.
+
+```ruby
+2.3.3 :003 > "hi".length
+#=> 2
+2.3.3 :004 > x = "hi"
+#=> "hi"
+2.3.3 :005 > x.length
+#=> 2
+```
+
+`Integer`s _do not_ have the `length` attribute.
+
+```ruby
+2.3.3 :010 > 3.14.length
+NoMethodError: undefined method 'length' for 3.14:Float
+```
+
+Just as fish don't have a "knit" method, `Float`s don't know how long they are.
+
+Attributes are tiny little facts that an instance knows about itself that it
+**_returns_** when asked.
+
+## Define "Dot-notation" Syntax
+
+We access an attribute by adding a `.` and then the `attribute name` after the
+instance or a variable pointing to the instance. We discovered `"hi"`'s class by
+asking it for its `.class` attribute.
+
+## Define Method
+
+Ruby instances have "internal" methods. You didn't define them by writing
+`def...end` somewhere. The makers of Ruby did! While attributes report simple
+facts, methods are activities that instances know how to do. In Ruby, a
+`String` knows how to **_return_** a `reverse`d or `capitalize`d version of
+itself.
+
+```ruby
+2.3.3 :011 > mirror = "olleh"
+#=> "olleh"
+2.3.3 :012 > mirror.reverse
+#=> "hello"
+2.3.3 :014 > mirror.capitalize
+#=> "Olleh"
+```
+
+Just as before, an `Integer` does not know how to `.reverse`:
+
+```ruby
+2.3.3 :013 > 2112.reverse
+NoMethodError: undefined method 'reverse' for 2112:Fixnum
+```
+
+**_What methods or attributes an instance has available depends on its class._**
+
+## Demonstrate Method Chaining
+
+Remember, expressions in Ruby always have a return value. And we just
+learned that methods and attributes _return_ things. This means we can chain
```

Comment:
I think these two sentences don't quite connect. Maybe show a simpler example that attributes always return something, and since we know that things always have attributes, we can sort of build a non chain example first, then the same example as a chain:

```
length_of_word = "hello".length
length_of_word
#=> 5

length_of_word.class
#=> Integer

"hello".length.class
#=> Integer
```

And even get weird/sort of self reinforcing with `"hello".length.class.class`